cunumeric/module.py

```diff
@@ -1637,10 +1756,7 @@ def vstack(tup: Sequence[ndarray]) -> ndarray:
     """
     # Reshape arrays in the `array_list` if needed before concatenation
     inputs = list(convert_to_cunumeric_ndarray(inp) for inp in tup)
-    reshaped = list(
-        inp.reshape([1, inp.shape[0]]) if inp.ndim == 1 else inp
-        for inp in inputs
-    )
+    reshaped = _atleast_nd(2, inputs, False)
```

Comment:
With this change the behavior of the function changes for singleton inputs, e.g.:

```
import cunumeric as cn
print(cn.vstack((cn.zeros((2,4)),)))
```

now returns:

```
[0. 0. 0. 0. 0. 0. 0. 0.]
```

whereas previously it returned:

```
[[0. 0. 0. 0.]
 [0. 0. 0. 0.]]
```

That is because `_atleast_nd` pulls out the first value in the case of singleton results, which isn't correct in this scenario.

Please fix this, and `dstack`, and make sure we test the singleton array case for those functions.